internal/service/ec2/vpc_endpoint_private_dns_verification.go

```diff
@@ -0,0 +1,105 @@
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource("aws_vpc_endpoint_service_private_dns_verification", name="Endpoint Service Private DNS Verification")
+func newResourceEndpointServicePrivateDNSVerification(_ context.Context) (resource.ResourceWithConfigure, error) {
+	return &resourceEndpointServicePrivateDNSVerification{}, nil
+}
+
+const (
+	ResNameEndpointServicePrivateDNSVerification = "Endpoint Service Private DNS Verification"
+)
+
+type resourceEndpointServicePrivateDNSVerification struct {
+	framework.ResourceWithConfigure
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_vpc_endpoint_service_private_dns_verification"
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"service_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().EC2Client(ctx)
+
+	var plan resourceEndpointServicePrivateDNSVerificationData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := &ec2.StartVpcEndpointServicePrivateDnsVerificationInput{
+		ServiceId: aws.String(plan.ServiceID.ValueString()),
+	}
+
+	out, err := conn.StartVpcEndpointServicePrivateDnsVerification(ctx, in)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.EC2, create.ErrActionCreating, ResNameEndpointServicePrivateDNSVerification, plan.ServiceID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	if out == nil || out.ReturnValue == nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.EC2, create.ErrActionCreating, ResNameEndpointServicePrivateDNSVerification, plan.ServiceID.String(), nil),
+			errors.New("empty output").Error(),
+		)
+		return
+	}
+	if !aws.ToBool(out.ReturnValue) {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.EC2, create.ErrActionCreating, ResNameEndpointServicePrivateDNSVerification, plan.ServiceID.String(), nil),
+			errors.New("request failed").Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Read is a no-op
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
Can embed `framework.WithNoUpdate` instead (verifies that `Update` is never called).